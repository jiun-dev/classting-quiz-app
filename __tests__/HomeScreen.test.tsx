import {StackNavigationProp} from '@react-navigation/stack';
import HomeScreen from '../src/screens/HomeScreen';
import {RouteProp} from '@react-navigation/native';
import {fireEvent, render} from '@testing-library/react-native';

describe('HomeScreen 컴포넌트 테스트', () => {
  const mockNavigate = jest.fn();
  const props: {
    navigation: StackNavigationProp<RootNavigationType, 'Home'>;
    route: RouteProp<RootNavigationType, 'Home'>;
  } = {
    navigation: {
      navigate: mockNavigate,
      goBack: jest.fn(),
      addListener: jest.fn(),
      removeListener: jest.fn(),
      setOptions: jest.fn(),
    } as any,
    route: {
      key: 'mockKey',
      name: 'Home',
      params: {},
    },
  };

  test('컴포넌트 스냅샷 테스트', () => {
    const rendered = render(<HomeScreen {...props} />);

    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  test('컴포넌트 렌더링 테스트', () => {
    const {getByText} = render(<HomeScreen {...props} />);
    const quizButtonElement = getByText('퀴즈 풀기');
    const wrongAnswerButtonElement = getByText('오답 노트 확인');

    expect(quizButtonElement).toBeDefined();
    expect(wrongAnswerButtonElement).toBeDefined();

    fireEvent.press(quizButtonElement);
    fireEvent.press(wrongAnswerButtonElement);

    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledTimes(2);
  });
});
