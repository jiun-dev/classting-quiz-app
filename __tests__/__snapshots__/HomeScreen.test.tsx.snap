// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen 컴포넌트 테스트 컴포넌트 렌더링 테스트 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#4CAF50",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
      }
    }
  >
    CLASSTING QUIZ
  </Text>
  <View
    style={
      {
        "marginVertical": 8,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4CAF50",
          "borderRadius": 16,
          "fontSize": 18,
          "marginVertical": 8,
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 22,
          }
        }
      >
        퀴즈 풀기
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 8,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4CAF50",
          "borderRadius": 16,
          "fontSize": 18,
          "marginVertical": 8,
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 22,
          }
        }
      >
        오답 노트 확인
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`HomeScreen 컴포넌트 테스트 컴포넌트 스냅샷 테스트 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#4CAF50",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 16,
      }
    }
  >
    CLASSTING QUIZ
  </Text>
  <View
    style={
      {
        "marginVertical": 8,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4CAF50",
          "borderRadius": 16,
          "fontSize": 18,
          "marginVertical": 8,
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 22,
          }
        }
      >
        퀴즈 풀기
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "marginVertical": 8,
        "width": "80%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4CAF50",
          "borderRadius": 16,
          "fontSize": 18,
          "marginVertical": 8,
          "opacity": 1,
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FFFFFF",
            "fontSize": 22,
          }
        }
      >
        오답 노트 확인
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
